cfme/services/requests.py

```diff
@@ -1,286 +1,370 @@
 # -*- coding: utf-8 -*-
 from contextlib import contextmanager
-from functools import partial
-from navmazing import NavigateToAttribute
+from navmazing import NavigateToAttribute, NavigateToSibling
+from widgetastic.widget import Text, Table, Checkbox
+from widgetastic_manageiq import BreadCrumb, SummaryFormItem, PaginationPane
+from widgetastic_patternfly import Button, View, Input, Tab, BootstrapTreeview, FlashMessages
 
+from cfme.base.login import BaseLoggedInPage
 from cfme.exceptions import RequestException
 from cfme.fixtures import pytest_selenium as sel
-from cfme.web_ui import (
-    Input, Region, Table, fill, flash, paginator, toolbar, match_location)
+from cfme.web_ui import paginator
 from utils.log import logger
 from utils.appliance import Navigatable
 from utils.appliance.implementations.ui import navigator, CFMENavigateStep, navigate_to
 
-REQUEST_FINISHED_STATES = {'Migrated', 'Finished'}
 
-buttons = Region(
-    locators=dict(
-        approve="//*[@title='Approve this Request']",
-        deny="//*[@title='Deny this Request']",
-        copy="//*[@title='Copy original Request']",
-        edit="//*[@title='Edit the original Request']",
-        delete="//*[@title='Delete this Request']",
-        submit="//span[@id='buttons_on']/a[@title='Submit']",
-        cancel="//a[@title='Cancel']",
-    )
-)
-
-request_table = Table('//*[@id="list_grid"]/table')
-fields = Region(
-    locators=dict(
-        reason=Input("reason"),
-        request_list=request_table
-    )
-)
-
-match_page = partial(match_location, controller='miq_request', title='Requests')
+class Request(Navigatable):
+    """
+    Class describes request row from Services - Requests page
+    """
 
+    REQUEST_FINISHED_STATES = {'Migrated', 'Finished'}
 
-# TODO Refactor these module methods and their callers for a proper request class
-class Request(Navigatable):
-    def __init__(self, appliance=None):
+    def __init__(self, row_description=None, cells=None, partial_check=False, appliance=None):
+        """
+        Args:
+            row_description: by default we'll be checking Description column to find required row
+            cells: cells used to find required row in table
+            partial_check: greedy search or not?
+        """
         Navigatable.__init__(self, appliance=appliance)
+        self.row_description = row_description
+        self.cells = {'Description': self.row_description} if cells is None else cells
+        self.partial_check = partial_check
+        self.view = navigate_to(self, 'All')
```

Comment:
wouldn't be better to use rest here to retrieve necessary data ? from my perspective navigation in __init__ doesn't look.